pkg/autodiscovery/README.md

```diff
@@ -1,15 +1,55 @@
 # package `autodiscovery`
 
-This package is a core piece of the agent. It is responsible for collecting check configurations from different sources (see package [config providers](https://github.com/DataDog/datadog-agent/tree/main/pkg/autodiscovery/providers)) and then schedule or unschedule integration configurations with the help of the schedulers.
+This package manages configuration for dynamic entities like pods and containers.
 
-It is also responsible for listening to container-related events and resolve template configurations that would match them.
+## Architecture
 
-## `AutoConfig`
+The high-level architecture of this package is implemented by the AutoConfig type, and looks like this:
 
-As a central component, `AutoConfig` owns and orchestrates several key modules:
+```
+               ┌────────────────┐
+               │ Workload Meta  │
+               └──┬──────────┬──┘
+  Static          │          │
+  Files──┐        │          │
+         │        │          │
+    ┌────▼────────▼──┐    ┌──▼─────────────┐
+    │Config Providers│    │   Listeners    │
+    └──┬─────┬───────┘    └────────┬───────┘
```

Comment:
Nit: there is another config source, the `clusterchecks` and `endpointschecks` config provider consumes configs from the datadog cluster agent.